active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,572 @@
+
+```xaml
+<Application.Resources>
+
+    <!--Color for error messages-->
+    <Color x:Key="SystemErrorColor">#CE3880</Color>
+    <SolidColorBrush x:Key="SystemControlErrorTextForegroundBrush" Value="SystemErrorColor"/>
+
+    <!--Brush for description messages-->
+    <SolidColorBrush x:Key="SystemControlDescriptionTextForegroundBrush" Value="SystemControlPageTextBaseMediumBrush"/>
+
+</Application.Resources>
+```
+
+## Change description text for validation
+
+Description text provides input to the end user on what format or content is expected from the control when it is not clear from the header and placeholder text.
+
+Developers can choose to either show the description text for a control or not. Default is that description text is off. 
+
+Description text should be used to complement headers and placeholders. Guidance on what each corresponds to is as follows: 
+1.	Headers give you an indication of what the control represents. It is used to catch the eye of a user scanning your form or to refer to a field.
+2.	Placeholders provide a succinct description of what should be input in the box. However, they aren’t an ideal pattern for guiding users input because user testing continually shows that placeholders in form fields often hurt usability more than help it. This is because: 
+    - Disappearing placeholder text strains users’ short-term memory.
+    -	Users may mistake a placeholder for data that was automatically filled in.
+    -	The default light-grey color of placeholder text has poor color contrast against most backgrounds.
+    -	Not all screen readers read placeholder text aloud
+    -	Often incompatible with autofill – you need to delete field content before finding the hint.
+3.	Description text is used to complement placeholder text. It provides more detailed formatting guidance that is needed while editing the field. 
+
+#### XAML
+```xaml
+    <TextBox 
+            x:Name="FirstName" 
+            Header="First Name:" 
+            Text="{x:Bind   ViewModel.Person.FirstName, 
+                            UpdateSourceTrigger=LostFocus, 
+                            Mode=TwoWay}"
+            DescriptionText="Use alphabetical characters (a-Z)."
+        />
+```
+
+#### C# (Model)
+```csharp
+    public class Person : ValidationWrapper
+    {
+        [Required]
+        [RegularExpression(@"^[a-zA-Z''-'\s]{1,40}$", ErrorMessage ="Use alphabetical characters (a-Z).")]
+        public string FirstName {
+            get { return GetValue<string>("FirstName"); }
+            set { SetValue("FirstName", value); }
+        }
+    }
+```
+
+## Validation for view-centric developers
+Code a view-centric (Winforms-like) developer would write to enable Validation.
+
+
+### 1.10.1	C# Sample
+#### XAML
+
+```xaml
+<TextBox x:Name="Email" 
+         Header="Email" 
+         Validation.IsRequired="True" 
```

Comment:
Does this indicate that the field is a required field? or that validation should be performed?

If this means that entry in this textbox is required
- How is that displayed?
- Can it be re-templated?
- How should a developer support this in custom controls?

Reply:
> How is that displayed?

By an '*' next to the header

> Can it be re-templated?

Yes, this will be part of the ControlTemplate

> How should a developer support this in custom controls?

By using the built-in controls as an example and following what they do

Reply:
I'm not sure that saying developers wishing to build their own custom controls should copy the built-in ones is adequate. 
Lots of developers will want to use C#/XAML, rather than the C++ the built-in ones are created with.
Hopefully toolkit controls will be updated accordingly and so they can be used as a reference by 3rd parties but may this also serve as a reminder that samples/examples often need to be in more than one language and copying C++ to create C# equivalents can be daunting for many.

Reply:
> Hopefully toolkit controls will be updated accordingly and so they can be used as a reference by 3rd parties but may this also serve as a reminder that samples/examples often need to be in more than one language and copying C++ to create C# equivalents can be daunting for many.

Fair point, thank you for bringing that up. I am working on a full end-to-end example for C# and will make sure it's made available